code/src/main/java/com/onlinecontacttracing/servlets/delete-data/DeleteDataAfterApproval.java

```diff
@@ -0,0 +1,52 @@
+package com.onlinecontacttracing.servlets;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.json.JsonFactory;
+import com.onlinecontacttracing.storage.PositiveUserContacts;
+import com.onlinecontacttracing.storage.PositiveUserPlaces;
+import java.security.GeneralSecurityException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServlet;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.logging.Logger;
+import static com.googlecode.objectify.ObjectifyService.ofy;
+
+@WebServlet("/delete-data-after-approval")
+public class DeleteDataAfterApproval extends HttpServlet {
+
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String CLIENT_ID = "83357506440-etvnksinbmnpj8eji6dk5ss0tbk9fq4g.apps.googleusercontent.com";
+  static final Logger log = Logger.getLogger(DeleteDataAfterApproval.class.getName());
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      // Set up
+      NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+      GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(httpTransport, JSON_FACTORY)
+        .setAudience(Collections.singletonList(CLIENT_ID))
+        .build();
+
+      // Get payload with userId
+      String idTokenString = request.getParameter("idToken");
+      GoogleIdToken idToken = verifier.verify(idTokenString);
+      Payload payload = idToken.getPayload();
+      String userId = payload.getSubject();
+
+      ofy().delete().type(PositiveUserContacts.class).id(userId).now();
```

Comment:
This is a very nitty thing, but I'd move this into a function called delete, so all the specifics are contained there, and you just pass in userId. Very much encapsulates that logic.